app/Http/Requests/StoreImageAnnotation.php

```diff
@@ -50,7 +50,59 @@ public function withValidator($validator)
         $validator->after(function ($validator) {
             if ($this->input('shape_id') === Shape::wholeFrameId()) {
                 $validator->errors()->add('shape_id', 'Image annotations cannot have shape WholeFrame.');
+                return;
             }
+
+            if (!$this->input('points')) {
+                $validator->errors()->add('points', 'Invalid shape. No points given.');
+                return;
+            }
```

Comment:
This is not needed because we already have the `required` validation rule for the attribute.